lib/fog/vsphere/requests/compute/create_vm.rb

```diff
@@ -300,7 +300,7 @@ def create_disk(disk, operation = :add, options = {})
           file_operation = options[:file_operation] || (:create if operation == :add)
           payload[:fileOperation] = file_operation if file_operation
 
-          if operation == :add && disk.thin == 'false' && disk.eager_zero == 'true'
+          if operation == :add && disk.thin == false && disk.eager_zero == true
```

Comment:
I know, I'm late to the party, but could you have just used `!disk.thin && disk.eager_zero` instead?

Reply:
Fixed with https://github.com/theforeman/foreman/pull/7634